hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1144,7 +1174,34 @@ def adaptive_example_deletion(self):
         If we do not make any successful changes, we recurse to the example's
         children and attempt the same there.
         """
-        self.example_wise_shrink(Length)
+        indices = [
+            (i, j)
+            for i, ls in enumerate(self.fine_grained_partitions)
+            for j in hrange(len(ls))
+        ]
+        self.random.shuffle(indices)
+
+        for i, j in indices:
+            if i >= len(self.fine_grained_partitions):
+                continue
+            partition = self.fine_grained_partitions[i]
+            # No point in trying to delete the last element because that will always
+            # give us a prefix.
+            if j >= len(partition) - 1:
+                continue
+
+            def clear_region(a, b):
+                assert a <= j <= b
+                if a < 0 or b >= len(partition) - 1:
+                    return False
+                return self.consider_new_buffer(
+                    self.buffer[: partition[a]] + self.buffer[partition[b] :]
+                )
+
+            to_right = find_integer(lambda n: clear_region(j, j + n))
+
+            if to_right > 0:
+                j -= find_integer(lambda n: clear_region(j - n, j + to_right))
```

Comment:
This `j -=` doesn't appear to do anything.

Reply:
Whoops, it was as part of recreating the logic for avoiding retrying the same deletions, but I couldn't convince myself it was actually useful and seemed to slightly slow down the tests so I got rid of it but left that vestigial bit.